src/vaadin-overlay.html

```diff
@@ -527,6 +527,7 @@
         }
 
         if (opened) {
+          this.__previouslyFocusedElement = document.activeElement;
```

Comment:
`document.activeElement` wouldn't work inside a shadow root. Need to use `getRootNode().activeElement`. Add a test for this case also.

Reply:
Done. Also, put the implementation behind the flag to make it opt-in.